lib/trino-memory-context/src/main/java/io/trino/memory/context/AggregatedMemoryContext.java

```diff
@@ -27,6 +27,8 @@ static AggregatedMemoryContext newRootAggregatedMemoryContext(MemoryReservationH
 
     AggregatedMemoryContext newAggregatedMemoryContext();
 
+    AggregatedMemoryContext newAggregatedMemoryContext(MemoryAllocationValidator memoryAllocationValidator);
```

Comment:
I wonder if this shouldn't be solved locally at task level via combination of `newRootAggregatedMemoryContext` and `MemoryReservationHandler`. Are there more use cases for `newAggregatedMemoryContext(MemoryAllocationValidator memoryAllocationValidator)` other than task allocation?

For example both `OperatorContext` and `WorkProcessorPipelineSourceOperator` both have their own `InternalAggregatedMemoryContext`. I'm not sure we should mix `MemoryAllocationValidator memoryAllocationValidator` into that picture (this adds complexity to the allocators structure)

Reply:
@sopel39 PTAL now